hazelcast/src/main/java/com/hazelcast/jet/impl/execution/init/ExecutionPlanBuilder.java

```diff
@@ -87,11 +86,25 @@ public static CompletableFuture<Map<MemberInfo, ExecutionPlan>> createExecutionP
         final VerticesIdAndOrder verticesIdAndOrder = VerticesIdAndOrder.assignVertexIds(dag);
         final int defaultParallelism = nodeEngine.getConfig().getJetConfig().getCooperativeThreadCount();
         final EdgeConfig defaultEdgeConfig = nodeEngine.getConfig().getJetConfig().getDefaultEdgeConfig();
-        Set<Integer> requiredPartitions = jobConfig.getArgument(KEY_REQUIRED_PARTITIONS);
+        final Set<Integer> requiredPartitions = jobConfig.getArgument(KEY_REQUIRED_PARTITIONS);
+
         final Map<MemberInfo, ExecutionPlan> plans = new HashMap<>();
         int memberIndex = 0;
 
         final Map<MemberInfo, int[]> partitionsByMember = getPartitionAssignment(nodeEngine, memberInfos, requiredPartitions);
+
+        if (!nodeEngine.getNode().isLiteMember()) {
+            Address localMemberAddress = nodeEngine.getThisAddress();
+            MemberInfo localMemberInfo = memberInfos.stream()
+                    .filter(mi -> mi.getAddress().equals(localMemberAddress))
+                    .findAny()
+                    .orElseThrow();
+            partitionsByMember.computeIfAbsent(localMemberInfo, (i) -> {
+                nodeEngine.getLogger(ExecutionPlanBuilder.class).info("Adding coordinator to partition-pruned job members");
```

Comment:
if we want to keep that log (do we?) it should not be on INFO level

Reply:
https://github.com/hazelcast/hazelcast/pull/24813/commits/96b0110c977e907bf2d85495319989801588cfd2